components/raftstore-v2/src/operation/ready/mod.rs

```diff
@@ -139,14 +198,18 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             // it may block for ever when there is unapplied conf change.
             self.set_has_ready();
         }
-        if committed_index > applied_index + PAUSE_FOR_RECOVERY_GAP {
+        if committed_index > applied_index + PAUSE_FOR_RECOVERY_GAP && let Some(w) = watch {
```

Comment:
This changed old behavior for peers created after restart. Not sure if it's good.

Reply:
It effectively does not change the behavior because in restart scenario, the watch must be present. And only when new peer is created from life.rs, the watch will be empty, which is OK as the peer is uninitialized and won't meet this  committed_index > applied_index + PAUSE_FOR_RECOVERY_GAP.
Maybe we can add an assert here.